### DISCUSSION TITLE
RAM requirements for jobs scheduled to HTCondor CEs

### DISCUSSION THREAD

**Discussion by arrabito** [DIRACGrid/DIRAC#6635] [Q&A] [answered]
Dear all,

I would like to know if there is anyway to specify RAM requirements for jobs scheduled to HTCondorCE.

I know how to do it with ARC CE, setting the requirement in the XRLS string, but I'm not sure if there is an equivalent for HTCondorCE.

Thank you.

Luisa

**Comment by sfayer** [comment, marked as answer]
Hi Luisa,

We do a similar thing for HTCondorCE, we set this option to request 4GB of RAM by default (along with the usual delegation lifetime flag), this should also work at a CE (or possibly even queue) level if needed:
```
Resources/Computing/HTCondorCE/ExtraSubmitString = +maxMemory = 4096 \n delegate_job_GSI_credentials_lifetime = 0
```

Regards,
Simon

**Comment by arrabito** [comment]
Thanks a lot.
I'll give it a try. I've never used however the delegation lifetime flag.
Can you please tell me in which cases is it needed?

**Comment by sfayer** [comment]
From what I remember, the lifetime flag is to stop the CE from reducing the pilot proxy lifetime to something like 24 hours by default... Setting it to 0 lets it use the full ~2 days that DIRAC sends: It may be an old hack that isn't needed any more, I'm not sure, we've not tested without it for a very long time!

Regards,
Simon

**Comment by arrabito** [comment]
OK thank you.

I will try without and let you know.
